runtime/onert/core/src/exec/Execution.test.cc

```diff
@@ -327,10 +332,74 @@ TEST(ExecInstance, simple)
   }
 }
 
+TEST(ExecInstance, shapeinf)
+{
+  auto mockup = MockUpModel();
+  mockup.compile();
+  auto executors = mockup.artifact->_executors;
+
+  auto input1 = IOIndex{0};
+  auto input2 = IOIndex{1};
+  auto output = IOIndex{0};
+
+  const onert::ir::Shape new_shape{2, 2, 2, 1};
+  const float input1_buffer[8] = {1, 0, -1, -2, 1, 2, 0, -1};
+  const float input2_buffer[8] = {1, -3, 2, -4, 4, -2, 3, 1};
+  float output_buffer[8] = {};
+  const float output_expected[8] = {5, -2, 0, -1, 8, 1, 2, 5};
+
+  onert::exec::Execution execution{executors};
+
+  execution.changeInputShape(input1, new_shape);
+  execution.changeInputShape(input2, new_shape);
+  execution.setInput(input1, reinterpret_cast<const void *>(input1_buffer), 32);
+  execution.setInput(input2, reinterpret_cast<const void *>(input2_buffer), 32);
+  execution.setOutput(output, reinterpret_cast<void *>(output_buffer), 32);
```

Comment:
(Optional) Using `sizeof` may be safer than hardcoding byte sizes because it automatically adapts to changes in array length or element type. 
```suggestion
  execution.setInput(input1, input1_buffer, sizeof(input1_buffer));
  execution.setInput(input2, input2_buffer, sizeof(input2_buffer));
  execution.setOutput(output, output_buffer, sizeof(output_buffer));
```